pysb/simulator/scipyode.py

```diff
@@ -376,38 +378,58 @@ def run(self, tspan=None, initials=None, param_values=None):
         trajectories = np.ndarray((n_sims, len(self.tspan),
                               len(self._model.species)))
-        for n in range(n_sims):
-            self._logger.info('Running simulation %d of %d', n + 1, n_sims)
+
+        if n_jobs == 1:
+            for n in range(n_sims):
+                self._logger.info('Running simulation %d of %d', n + 1, n_sims)
+                if self.integrator == 'lsoda':
+                    trajectories[n] = scipy.integrate.odeint(
+                        self.func,
+                        self.initials[n],
+                        self.tspan,
+                        Dfun=self.jac_fn,
+                        args=(self.param_values[n],),
+                        **self.opts)
+                else:
+                    self.integrator.set_initial_value(self.initials[n],
+                                                      self.tspan[0])
+                    # Set parameter vectors for RHS func and Jacobian
+                    self.integrator.set_f_params(self.param_values[n])
+                    if self._use_analytic_jacobian:
+                        self.integrator.set_jac_params(self.param_values[n])
+                    trajectories[n][0] = self.initials[n]
+                    i = 1
+                    while self.integrator.successful() and self.integrator.t < \
+                            self.tspan[-1]:
+                        self._logger.log(EXTENDED_DEBUG,
+                                         'Simulation %d/%d Integrating t=%g',
+                                         n + 1, n_sims, self.integrator.t)
+                        trajectories[n][i] = self.integrator.integrate(self.tspan[i])
+                        i += 1
+                    if self.integrator.t < self.tspan[-1]:
+                        trajectories[n, i:, :] = 'nan'
+        else:
             if self.integrator == 'lsoda':
-                trajectories[n] = scipy.integrate.odeint(
-                    self.func,
-                    self.initials[n],
-                    self.tspan,
-                    Dfun=self.jac_fn,
-                    args=(self.param_values[n],),
-                    **self.opts)
+                p = Pool(processes=n_jobs)
+
+                def worker(*args, **kwargs):
+                    func, y0, t = args
+                    Dfun, args = kwargs.values()
+                    trajectory = scipy.integrate.odeint(func, y0, t, Dfun=Dfun, args=args)
+                    return trajectory
+
+                results = [p.apply_async(worker, (self.func, self.initials[i], self.tspan),
+                                         {'Dfun': self.jac_fn, 'args': (self.param_values[i],)}) for i in range(n_sims)]
+                p.close()
+                p.join()
+                trajectories = [tr.get() for tr in results]
             else:
-                self.integrator.set_initial_value(self.initials[n],
-                                                  self.tspan[0])
```

Comment:
You might want to check out the new ode integrate options. They simplified all solvers into an "odeint" like call (not requiring calls to step). 
https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html#scipy.integrate.solve_ivp

It adds a few more integrators for no more cost. I saw this awhile back and was planning on doing it but havent had time yet.

Reply:
I added the solvers from scipy.integrate.solve_ivp, would you mind taking a look at the code?